platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/ActionHandlerServiceImpl.java

```diff
@@ -98,6 +114,19 @@ public void delete(Map<String, Object> parameters) throws ActionHandlerException
         dataService.delete(instance);
     }
 
+    private void setInstancePropertiesForQuery(Object instance, List<Field> fields, Map<String, Object> properties) throws ActionHandlerException {
+        for (Field field : fields) {
+            Object value = properties.get(field.getName());
+            if (value instanceof List) {
```

Comment:
You can put it all into one condition and use CollectionUtils.isNotEmpty() on a casted _value_ object.